common/stream-ratelimiter/src/lib.rs

```diff
@@ -76,6 +77,12 @@ impl<T: Stream> Stream for RateLimiter<T> {
     fn poll_next(mut self: Pin<&mut Self>, cx: &mut task::Context<'_>) -> Poll<Option<Self::Item>> {
         // If the interval is ready, add permits. This is safe to do in all calls to poll_next().
         if let Poll::Ready(Some(_)) = self.as_mut().project().interval.poll_next(cx) {
+            if self.permits.available_permits() == 0 {
```

Comment:
I didn't review this code, but why this here versus at line 84/91 where we didn't acquire a permit?

Reply:
didn't review this code either... also I think L100 (not from this PR) is a memory leak 😬 

tbh not totally sure this code is correct...

Reply:
one issue with this warn log is we have no context here (namely peer_id)